### PR TITLE
Clean-up CI concerns: Separate what from how

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Just open for discussion.

**Comment by sjrd** [comment]
That makes sense. Is the content of `ci/scalajs-task-worker.sh` meant to be inlined in the Jenkins script? Or should we change the Jenkins script to call `ci/scalajs-task-worker.sh`?

**Comment by gzm0** [comment]
ci/scalajs-task-worker.sh is supposed to be inlined. Adding to VC only since it is growing larger.

**Comment by sjrd** [comment]
OK looks good. I have updated the script on Jenkins. Could you amend the commit message to something a bit less "proposal"?

**Comment by gzm0** [comment]
Updated. Review by @sjrd

**Comment by sjrd** [comment]
LGTM
(not retrying now because GitHub is under DDoS)

**Comment by sjrd** [comment]
retest this please

**Comment by sjrd** [comment]
retest this please

File: ci/jenkins-track/scalajs-task-worker.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    "-Djline.terminal=jline.UnsupportedTerminal" \
+    "-Dsbt.boot.directory=$LOC_SBT_BOOT" \
+    "-Dsbt.ivy.home=$LOC_SBT_HOME" \
+    "-Divy.home=$LOC_SBT_HOME" \
+    "-Dsbt.global.base=$LOC_SBT_BASE" \
+    '"' >> tmp.sh
+
+# Define setJavaVersion
+
+echo >> tmp.sh <<EOF
```
Comment (sjrd) [inline ci/jenkins-track/scalajs-task-worker.sh]: Apparently this syntax does not work.

**Comment by gzm0** [comment]
Updated

File: ci/jenkins-track/scalajs-task-worker.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+echo 'setJavaVersion() {' \
+     '  export JAVA_HOME=$HOME/apps/java-$1' \
+     '}' >> tmp.sh
+
+# Define sbt and sbtretry
+
+echo 'sbtretry() {' \
+     '  local TIMEOUT=25m' \
+     '  echo "RUNNING timeout -k 5 $TIMEOUT sbt" "$@"' \
```
Comment (sjrd) [inline ci/jenkins-track/scalajs-task-worker.sh]: This won't actually print new lines in the file. And a shell script entirely on one line, doesn't work.

**Comment by sjrd** [comment]
retest this please

File: ci/jenkins-track/scalajs-task-worker.sh (hunk 2 above)
Comment (gzm0) [inline ci/jenkins-track/scalajs-task-worker.sh]: Argh... Stupid me... And with the `'` quotes I cannot write `\n`... And if I use `"` quotes, stuff gets interpolated... :(

Why is it I cry every time I have to deal with Jenkins?

Comment (gzm0) [inline ci/jenkins-track/scalajs-task-worker.sh]: Well... It's going to be the I/O inefficient solution :)

**Comment by gzm0** [comment]
Updated. You're gonna love this one `C-x r t` FTW

**Comment by sjrd** [comment]
LGTM. retest this please

**Pull request merged**